adbms/src/main/scala/de/up/hpi/informationsystems/adbms/definition/RowRelation.scala

```diff
@@ -1,12 +1,26 @@
 package de.up.hpi.informationsystems.adbms.definition
+import de.up.hpi.informationsystems.adbms.definition
+
 import scala.util.Try
 
 abstract class RowRelation extends Relation {
 
   private var data: Seq[Record] = Seq.empty
 
   /** @inheritdoc */
-  override def insert(record: Record): Unit = data = data :+ record
+  override def insert(record: Record): Try[Record] = Try(internal_insert(record))
+
+  @throws[IncompatibleColumnDefinitionException]
+  private def internal_insert(record: Record): Record =
+  // check for correct column layout
+    record.columns.equals((this.columns)) match {
+      case true => {
+        data = data :+ record
+        record
+      }
+      case false => throw IncompatibleColumnDefinitionException(s"this records column layout does not match this " +
+        s"relations schema:\n${record.columns} (record)\n${this.columns} (relation)")
+    }
```

Comment:
_minor style-remark_:

- parentheses in line 16 are redundant
- simple match-pattern can be replaced with if-else, see:
```scala
  @throws[IncompatibleColumnDefinitionException]
  private def internal_insert(record: Record): Record =
    // check for correct column layout
    if(record.columns == columns) {
      data = data :+ record
      record
    } else {
      throw IncompatibleColumnDefinitionException(s"this records column layout does not match this " +
        s"relations schema:\n${record.columns} (record)\n${this.columns} (relation)")
    }
```
(I would use if-else, but use what you prefer)